### PR TITLE
Add signal query argument for Restart Container command

### PR DESCRIPTION
**Pull request by dmmax**
Linked to the issue: https://github.com/docker-java/docker-java/issues/2123

### CODE REVIEW

**Comment by eddumelendez** [comment]
Can you add a test, please?

**Comment by dmmax** [comment]
Hi @eddumelendez 

I added an integration test that covers the new functionality. 

I also had to add new API versions to the `RemoteApiVersion`. Let me know if this should be rolled back (I can use a string argument instead)

**Comment by dmmax** [comment]
Hi @eddumelendez 
Can you have a look again into the PR? 🙏

File: docker-java/src/test/java/com/github/dockerjava/cmd/RestartContainerCmdImplIT.java [hunk 1]
```diff
@@ -44,6 +49,43 @@ public void restartContainer() throws DockerException {
...
     }
 
+    @Test
+    public void restartContainerWithSignal() throws Exception {
+        DefaultDockerClientConfig dockerClientConfig = DefaultDockerClientConfig.createDefaultConfigBuilder()
+            .withApiVersion(RemoteApiVersion.VERSION_1_42)
+            .withRegistryUrl("https://index.docker.io/v1/")
+            .build();
+        try (DockerClient dockerClient = createDockerClient(dockerClientConfig)) {
+            if (!getVersion(dockerClient).isGreaterOrEqual(RemoteApiVersion.VERSION_1_42)) {
```
Comment (eddumelendez) [inline docker-java/src/test/java/com/github/dockerjava/cmd/RestartContainerCmdImplIT.java:59]: I think we can use [Assume](https://junit.org/junit4/javadoc/4.12/org/junit/Assume.html) instead.

**Review by eddumelendez — APPROVED** [review]

**Comment by eddumelendez** [comment]
Thanks for your contribution, @dmmax !

**Pull request merged**